### ISSUE TITLE
Stack overflow when scrolling ListBox with changing content

### ISSUE THREAD

**Issue by ahopper** [AvaloniaUI/Avalonia#7706]
**Describe the bug**
When scrolling a ListBox and the content is changed in code at the same time a stack overflow results

**To Reproduce**
Run this  https://github.com/ahopper/ListBoxScrollStackoverflowRepro and scroll the listbox continuously, after 8s the overflow will probably happen

**Expected behavior**
No stack overflow.

**Desktop (please complete the following information):**

- OS: Windows
- Version 0.10.12 and 0.10.999-cibuild0018930-beta

**Comment by ahopper** [comment]
The same thing happens if the ListBox is replaced with
```
<ScrollViewer>
	<ItemsControl Items="{Binding Stuff}"/>
</ScrollViewer>
```

**Comment by grokys** [comment]
Partially fixed by #9677: stack overflow is gone but now the `ScrollBar` gets stuck.
